### PR TITLE
chore(github): adding storybook build workflow

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

To avoid breaking storybook build when updating dependencies (ref https://github.com/containers/podman-desktop/pull/7486#pullrequestreview-2101408827), adding a workflow for building storybook.

### How to test this PR?

- Pipeline should be ✅

### CODE REVIEW

File: .github/workflows/pr-check-storybook.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Execute yarn
+        run: yarn --frozen-lockfile
+
+      - name: Run website
```
Comment (benoitf) [inline .github/workflows/pr-check-storybook.yaml]: looks like a copy/paste issue

about the workflow I don't know how much it'll take, to know if we could not add it as part of the linters/checkers tests

File: .github/workflows/pr-check-storybook.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+name: Build storybook on pull request
+
+on:
+  pull_request:
+    paths:
+      - 'package.json'
+      - 'yarn.lock'
+      - 'storybook/**'
```
Comment (benoitf) [inline .github/workflows/pr-check-storybook.yaml:25]: I would say that you want to check storybook if you modify the components as well
(let say I remove Button component or move it elsewhere I would hope that PR check fails)

Comment (axel7083) [reply .github/workflows/pr-check-storybook.yaml:25]: Oh yeah smart

**Comment by axel7083** [comment]
@benoitf I changed the `storybook:build` to build the ui package before building storybook, as it depends on it. I also added as suggested the ui package to the trigger path.

File: .github/workflows/pr-check-storybook.yaml (hunk 2 above)
Comment (benoitf) [reply .github/workflows/pr-check-storybook.yaml:25]: maybe just add an entry https://github.com/containers/podman-desktop/blob/main/.github/workflows/pr-check.yaml#L176-L221 if it's taking only few seconds

**Review by benoitf — APPROVED** [review]
~~we can add the~~ (wrong text :-)

File: .github/workflows/pr-check-storybook.yaml (hunk 2 above)
Comment (axel7083) [reply .github/workflows/pr-check-storybook.yaml:25]: We need to build the ui package first, the linter would not have the packages build I guess ?

File: .github/workflows/pr-check-storybook.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      - 'packages/ui/src/**'
+
+jobs:
+  storybook-build:
+    name: build storybook
+    runs-on: ubuntu-22.04
```
Comment (benoitf) [inline .github/workflows/pr-check-storybook.yaml]:
```suggestion
    runs-on: ubuntu-24.04
```

**Pull request merged**